Sensus.Shared/Callbacks/CallbackScheduler.cs

```diff
@@ -124,7 +124,21 @@ public virtual Task RaiseCallbackAsync(string callbackId, bool repeating, int re
                                     SensusServiceHelper.Get().Logger.Log("Raising callback \"" + scheduledCallback.Id + "\".", LoggingLevel.Normal, GetType());
 
                                     if (notifyUser)
-                                        SensusContext.Current.Notifier.IssueNotificationAsync("Sensus", scheduledCallback.UserNotificationMessage, callbackId, true, scheduledCallback.DisplayPage);
+                                    {
+                                        bool playSound = true;
+
+                                        var runningProtocol = SensusServiceHelper.Get().GetRunningProtocolById(scheduledCallback.ProtocolId);
+
+                                        // check whether the current time is within any of the callback's protocol's notification sound exclusion windows.
+                                        // if we are within a window, don't play a sound.
+                                        if (runningProtocol != null)
+                                            foreach (Window window in runningProtocol.NotificationAlertExclusionWindowsList)
+                                                if (window.EncompassesTime(DateTime.Now))
+                                                    playSound = false;
+
+                                        SensusContext.Current.Notifier.IssueNotificationAsync("Sensus", scheduledCallback.UserNotificationMessage, callbackId, scheduledCallback.ProtocolId, playSound, scheduledCallback.DisplayPage);
```

Comment:
Make the exclusion window checking method `public static` and call that method here instead of duplicating code.